discovery-infra/start_discovery.py

```diff
@@ -77,13 +79,16 @@ def create_nodes_and_wait_till_registered(inventory_client, cluster, image_path,
 # If master in name -> role will be master, same for worker
 def set_hosts_roles(client, cluster_id):
     hosts = []
-    libvirt_macs = utils.get_libvirt_nodes_mac_role_ip_and_name()
+    libvirt_nodes = utils.get_libvirt_nodes_mac_role_ip_and_name()
     inventory_hosts = client.get_cluster_hosts(cluster_id)
-    assert len(libvirt_macs) == len(inventory_hosts)
-    for host in inventory_hosts:
-        hw = json.loads(host["hardware_info"])
-        role = [libvirt_macs[nic["mac"]]["role"] for nic in hw["nics"] if nic["mac"].lower() in libvirt_macs][0]
-        hosts.append({"id": host["id"], "role": role})
+
+    for libvirt_mac, libvirt_metadata in libvirt_nodes.items():
```

Comment:
The main loop was going over the hosts and I wanted the workflow logic to follow the libvirt nodes.
Because you don't care about all of the hosts - Your main issue is with your nodes.

Reply:
> What is this PR about? Please add a description and/or JIRA ticket.

Description was added.